### PR TITLE
sending Heimdal events in a batch

### PR DESCRIPTION
**Pull request by eishay**
/cc @stkem

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/heimdal/HeimdalServiceClient.scala [hunk 1]
```diff
@@ -29,18 +36,60 @@ trait HeimdalServiceClient extends ServiceClient {
...
     val airbrakeNotifier: AirbrakeNotifier,
     val httpClient: HttpClient,
     serviceDiscovery: ServiceDiscovery
   ) extends FortyTwoActor(airbrakeNotifier) with ServiceClient with Logging {
 
-  final val serviceType = ServiceType.HEIMDAL
+  private final val serviceType = ServiceType.HEIMDAL
   val serviceCluster = serviceDiscovery.serviceCluster(serviceType)
+  private var events: Vector[UserEvent] = Vector()
+  private var closing = false
```
Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/heimdal/HeimdalServiceClient.scala:61]: Is this to force flush on every event as the service is going down? It seems not to be set to `true` anywhere.

File: kifi-backend/modules/common/app/com/keepit/heimdal/HeimdalServiceClient.scala [hunk 2]
```diff
@@ -29,18 +36,60 @@ trait HeimdalServiceClient extends ServiceClient {
...
+      } else {
+        events.size match {
+          case s if(s >= EventQueueSize.LowWatermarkBatchSize) =>
+            self ! FlushEventQueue //flush with the events in the actor mailbox
+          case s if(s >= EventQueueSize.MaxBatchSize) =>
+            flush() //flushing without taking in account events in the mailbox
+        }
+      }
+    case FlushEventQueueAndClose =>
+      closing = false
```
Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/heimdal/HeimdalServiceClient.scala]: should be `closing = true``?

File: kifi-backend/modules/common/app/com/keepit/heimdal/HeimdalServiceClient.scala [hunk 3]
```diff
@@ -29,18 +36,60 @@ trait HeimdalServiceClient extends ServiceClient {
...
 
   def receive = {
-    case event: UserEvent => call(Heimdal.internal.trackEvent, Json.toJson(event))
-    case m => throw new UnsupportedActorMessage(m)
+    case event: UserEvent =>
+      events = events :+ event
+      if (closing) {
+        flush()
+      } else {
+        events.size match {
```
Comment (stkem) [inline kifi-backend/modules/common/app/com/keepit/heimdal/HeimdalServiceClient.scala:70]: FlushEventQueue just calls `flush()`. Whats the difference in the two cases here?

File: kifi-backend/modules/common/app/com/keepit/heimdal/HeimdalServiceClient.scala (hunk 2 above)
Comment (eishay) [inline kifi-backend/modules/common/app/com/keepit/heimdal/HeimdalServiceClient.scala]: yes!

**Pull request merged**